src/EnrollOnPremController.js

```diff
@@ -35,9 +35,13 @@ function (Okta, FormType, FormController, Footer, TextBox) {
     },
     Model: function () {
       var provider = this.options.provider;
+      var factors = this.options.appState.get('factors');
+      var factor = factors.findWhere(_.pick(this.options, 'provider', 'factorType'));
+      var profile = factor.get('profile');
```

Comment:
Is it guaranteed that every defined factor has a ``profile``? Just making sure we don't hit an undefined error in line 41 in case ``profile`` turns out to be ``undefined``.